File: VB/Data/Northwind/Category.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations.Schema

Namespace DataMappings.Data.Northwind

    Public Partial Class Category

        Public Sub New()
            Products = New HashSet(Of Product)()
        End Sub

        Public Property CategoryId As Integer

        Public Property CategoryName As String

        Public Property Description As String

        Public Property Picture As Byte()

        <NotMapped>
        Public Property HasProducts As Boolean = True

        Public Overridable Property Products As ICollection(Of Product)
    End Class
End Namespace

File: CS/Data/Northwind/Category.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

#nullable disable

namespace DataMappings.Data.Northwind {
    public partial class Category {
        public Category() {
            Products = new HashSet<Product>();
        }

        public int CategoryId { get; set; }
        public string CategoryName { get; set; }
        public string Description { get; set; }
        public byte[] Picture { get; set; }

        [NotMapped]
        public bool HasProducts { get; set; } = true;

        public virtual ICollection<Product> Products { get; set; }
    }
}
